### PR TITLE
add DAO to check DBS PhEDEx status

### PR DESCRIPTION
**Pull request by ticoann**
initial patch for getting DBS PhEDEx upload status from wmbs data.

still need to add 
1. AgentDataCollector need to add this info to wmstats report. (done)
2. reqmgr2/or wmstats collect information and advance the status (will do this in the separate patch)
3. need to add unittest

### CODE REVIEW

File: src/python/WMComponent/AnalyticsDataCollector/AnalyticsPoller.py [hunk 1]
```diff
@@ -94,6 +97,9 @@ def algorithm(self, parameters):
             
             logging.info("Getting Finished Task Data ...")
             finishedTasks = self.wmagentDB.getFinishedSubscriptionByTask()
+            
+            logging.info("Getting DBS PhEDEx upload status ...")
+            completedWfs = self.dbsBufferUtil.getPhEDxDBSStatuForCompletedWorkflows(summary=True)
```
Comment (amaltaro) [inline src/python/WMComponent/AnalyticsDataCollector/AnalyticsPoller.py]: wrong method name

File: bin/check-phedex-dbs-status [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from __future__ import (print_function, division)
+
+from WMComponent.DBS3Buffer.DBSBufferUtil  import DBSBufferUtil
+from WMCore.WMInit import connectToDB
+from pprint import pprint
+
+if __name__ == "__main__":
+    connectToDB()
+    dbsUtil = DBSBufferUtil()
+    result = dbsUtil.getPhEDxDBSStatuForCompletedWorkflows(summary=True)
```
Comment (amaltaro) [inline bin/check-phedex-dbs-status]: wrong method name

File: src/python/WMComponent/DBS3Buffer/MySQL/CheckStatusForCompletedWorkflows.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            resultByWorkflow.setdefault(workflow, {})
+            resultByWorkflow[workflow].setdefault("InDBS", 0)
+            resultByWorkflow[workflow].setdefault("NotInDBS", 0)
+            resultByWorkflow[workflow].setdefault("InPhEDEx", 0)
+            resultByWorkflow[workflow].setdefault("NotInPhEDEx", 0)
+            resultByWorkflow[workflow].setdefault("EmptyTasks", 0)
+            
+            count = info["count"]
+            
+            if info['status'] == "InDBS":
```
Comment (amaltaro) [inline src/python/WMComponent/DBS3Buffer/MySQL/CheckStatusForCompletedWorkflows.py]: GLOBAL seems to be another popular state. This is something I want to investigate first to see whether we really need these 3 DBS status and what exactly is their meaning.

**Review by amaltaro — CHANGES_REQUESTED** [review]
I still fail to see how we are going to use this information, though. It's good to know that the agent has already injected all the data into DBS/PhEDEx, but I'm almost sure Ops would still fire calls to DBS/PhEDEx to run other checks.

Do we actually use any of this combined information at the moment?

File: src/python/WMComponent/AnalyticsDataCollector/AnalyticsPoller.py [hunk 4]
```diff
@@ -111,12 +117,14 @@ def algorithm(self, parameters):
                                    WorkflowsWithSkippedFile(%s), 
                                    Batch Job(%s),
                                    Finished Tasks(%s),
-                                   Local Queue(%s)  ...""" 
+                                   Local Queue(%s)  
+                                   Complted workflows(%s)..."""
```
Comment (ericvaandering) [inline src/python/WMComponent/AnalyticsDataCollector/AnalyticsPoller.py]: Misspelled

**Comment by ticoann** [comment]
replaced by #7757

**Pull request closed without merging**